certora/specs/AllowancesInvariant.spec

```diff
@@ -0,0 +1,95 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+using GeneralAdapter1 as GeneralAdapter1;
+using EthereumGeneralAdapter1 as EthereumGeneralAdapter1;
+using ParaswapAdapter as ParaswapAdapter;
+
+methods {
+    function _.approve(address token, address spender, uint256 amount)  external => summaryApprove(calledContract, spender, amount) expect bool;
+    // Aave dispatch
+    function _.repay(address, uint256, uint256, address) external => HAVOC_ECF;
+    // Compound dispatch
+    function _.repayBorrowBehalf(address, uint256) external => HAVOC_ECF;
+    function _.underlying() external => mockUnderlying() expect address;
+    function _.supplyTo(address, address, uint256) external => HAVOC_ECF;
+    function _.baseToken() external => mockBaseToken() expect address;
+    function _.compoundV2RedeemEth(uint256,address) external => HAVOC_ECF;
+    // Paraswap dispatch
+    function _.set(bytes memory, uint256 offset, uint256) internal => setData(offset) expect void;
+    function _.get(bytes memory, uint256 offset) internal => getData(offset) expect uint256;
+    function _.isValidAugustus(address) external => NONDET;
```

Comment:
same, because this is a view call it should be summarized as NONDET automatically

Reply:
[Run](https://prover.certora.com/output/66603/0d002a86adca4e6db01619b27c98dd02?anonymousKey=9c2c3bc50f72fad0f6a403e735cd771f694d172a) without either HAVOC_ECF and NONDET summaries